subiquity/server/controllers/install.py

```diff
@@ -306,6 +306,36 @@ async def run_curtin_step(name, stages, step_config, source=None):
                 source=source,
                 )
             await self.setup_target(context=context)
+
+            # For OEM, we basically mimic what ubuntu-drivers does:
+            # 1. Install each package with apt-get install
+            # 2. For each package, run apt-get update using only the source
+            # installed by said package.
+            # 3. Run apt-get install again for each package. This will upgrade
+            # them to the version found in the OEM archive.
+
+            # NOTE In ubuntu-drivers, this is done in a single call to apt-get
+            # install.
+            for pkg in self.model.oem.metapkgs:
+                await self.install_package(package=pkg)
+
+            if self.model.network.has_network:
+                for pkg in self.model.oem.metapkgs:
+                    source_list = f"/etc/apt/sources.list.d/{pkg}.list"
```

Comment:
I was going to ask if we can be sure that this is what the sources list entry is called, but I see this code is from ubuntu-drivers so presumably yes!

Reply:
Yes, I'm not sure either. I originally wanted to parse the output of `dpkg -L` or similar but I ended up taking the ubuntu-drivers approach instead.